### PR TITLE
HDDS-9080. om roles CLI should provide Json output option

### PR DESCRIPTION
**Pull request by Tejaskriya**
## What changes were proposed in this pull request?

Add the option for the command `ozone admin om roles` to output as JSON

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-9080

## How was this patch tested?

Added new Robot test. Also verified the JSON out using a docker-compose cluster:
```
ozone admin om roles --service-id=omservice --json
[ {
  "om1" : {
    "serverRole" : "FOLLOWER",
    "hostname" : "om1"
  }
}, {
  "om2" : {
    "serverRole" : "FOLLOWER",
    "hostname" : "om2"
  }
}, {
  "om3" : {
    "serverRole" : "LEADER",
    "hostname" : "om3"
  }
} ]
```

### CODE REVIEW

**Comment by jyotirmoy-gh** [comment]
Please correct the PR title `ozone admin cert list should provide JSON output option` -> `ozone admin om roles should provide JSON output option`

**Review by jyotirmoy-gh — CHANGES_REQUESTED** [review]
Please correct PR title - `ozone admin cert list should provide JSON output option` -> `ozone admin om roles should provide JSON output option`

**Comment by Tejaskriya** [comment]
> Please correct PR title - `ozone admin cert list should provide JSON output option` -> `ozone admin om roles should provide JSON output option`

Sorry for the eror

> Please correct PR title - `ozone admin cert list should provide JSON output option` -> `ozone admin om roles should provide JSON output option`

Sorry for the error, I have fixed it now

File: hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/admin/om/GetServiceRolesSubcommand.java [hunk 1]
```diff
@@ -76,4 +90,24 @@ private void getOmServerRoles(List<ServiceInfo> serviceList) {
...
+
+  private void getOmServerRolesAsJson(List<ServiceInfo> serviceList)
+      throws IOException {
+    List<Map<String, Map<String, String>>> omServiceList = new ArrayList<>();
+    for (ServiceInfo serviceInfo : serviceList) {
+      OMRoleInfo omRoleInfo = serviceInfo.getOmRoleInfo();
+      if (omRoleInfo != null &&
+          serviceInfo.getNodeType() == HddsProtos.NodeType.OM) {
+        Map<String, Map<String, String>> omService = new HashMap<>();
+        omService.put(serviceInfo.getOmRoleInfo().getNodeId(),
```
Comment (ashishkumar50) [inline hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/admin/om/GetServiceRolesSubcommand.java:102]:
```suggestion
        omService.put(omRoleInfo.getNodeId(),
```

File: hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/admin/om/GetServiceRolesSubcommand.java [hunk 2]
```diff
@@ -76,4 +90,24 @@ private void getOmServerRoles(List<ServiceInfo> serviceList) {
...
+            new HashMap<String, String>() {{
+              put("serverRole", serviceInfo.getOmRoleInfo().getServerRole());
```
Comment (ashishkumar50) [inline hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/admin/om/GetServiceRolesSubcommand.java:104]:
```suggestion
              put("serverRole", omRoleInfo.getServerRole());
```

**Review by ashishkumar50 — COMMENTED** [review]
@Tejaskriya thanks for working on this. Change LGTM, given two suggestions.

File: hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/admin/om/GetServiceRolesSubcommand.java (hunk 2 above)
Comment (Tejaskriya) [reply hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/admin/om/GetServiceRolesSubcommand.java]: I have incorporated your suggestions in the new method and the old one too. Thank you.

File: hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/admin/om/GetServiceRolesSubcommand.java [hunk 3]
```diff
@@ -70,10 +84,30 @@ private void getOmServerRoles(List<ServiceInfo> serviceList) {
...
-            serviceInfo.getOmRoleInfo().getNodeId() + " : " +
-                serviceInfo.getOmRoleInfo().getServerRole() + " (" +
+            omRoleInfo.getNodeId() + " : " +
+                omRoleInfo.getServerRole() + " (" +
                 serviceInfo.getHostname() + ")");
       }
     }
   }
+
+  private void getOmServerRolesAsJson(List<ServiceInfo> serviceList)
```
Comment (sadanand48) [inline hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/admin/om/GetServiceRolesSubcommand.java:94]: minor nit:  `printOmServerRolesAsJson` fits better. same for above method too

Comment (Tejaskriya) [reply hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/admin/om/GetServiceRolesSubcommand.java]: Thank you for the suggestion, I have taken it up in the next commit

**Review by sadanand48 — APPROVED** [review]

**Pull request merged**